runtime/src/rent_collector.rs

```diff
@@ -83,19 +83,26 @@ impl RentCollector {
     /// given an account that 'should_collect_rent'
     /// returns (amount rent due, is_exempt_from_rent)
     pub fn get_rent_due(&self, account: &impl ReadableAccount) -> RentDue {
-        let slots_elapsed: u64 = (account.rent_epoch()..=self.epoch)
-            .map(|epoch| self.epoch_schedule.get_slots_in_epoch(epoch + 1))
-            .sum();
-
-        // avoid infinite rent in rust 1.45
-        let years_elapsed = if self.slots_per_year != 0.0 {
-            slots_elapsed as f64 / self.slots_per_year
+        if self
+            .rent
+            .is_exempt(account.lamports(), account.data().len())
+        {
+            RentDue::Exempt
         } else {
-            0.0
-        };
-
-        self.rent
-            .due(account.lamports(), account.data().len(), years_elapsed)
+            let slots_elapsed: u64 = (account.rent_epoch()..=self.epoch)
+                .map(|epoch| self.epoch_schedule.get_slots_in_epoch(epoch + 1))
+                .sum();
+
+            // avoid infinite rent in rust 1.45
+            let years_elapsed = if self.slots_per_year != 0.0 {
+                slots_elapsed as f64 / self.slots_per_year
+            } else {
+                0.0
+            };
+
+            // we know this account is not exempt
+            self.rent.paying_due(account.data().len(), years_elapsed)
+        }
```

Comment:
I like this idea! Do you have actual numbers/benchmarks to confirm this does in fact improve performance?

I ask since it does add a branch (that hopefully is predicted correctly, assuming most accounts are rent exempt), which could negatively impact performance.

Reply:
Oh, yuck, sorry for the way this comment shows up. I meant to just select the new impl of `fn get_rent_due()`.

Reply:
I don't believe it adds a branch. It just moves the existing branch earlier in the function.

Reply:
The calculation that we are running is order # accounts per epoch. Eliminating this work is certainly a net positive. This code runs each time we collect rent for an account in the rent partition. This code also runs each time we load an account for a transaction. It also runs for each account at startup for generating the initial index. It will also run during hash calculation once per account when we eliminate rewrites.

Reply:
Yep you're right, sorry, I wasn't looking at the bigger context here (re: adding/moving branch).